### ISSUE TITLE
TypeError:  x-original-urlis not a legal HTTP header value

### ISSUE THREAD

**Issue by AppleChill** [apollographql/apollo-server#4086]
Hi everyone.
I am uing apollo-server-azure-functions in azure functions, it was working until 5/7 ,but it is not working now.
I wrote code and deploy function again, but it didn't work,I am not sure why I am getting this error.
Does anyone has seen this error same like me?

![截圖 2020-05-08 下午4 53 17](https://user-images.githubusercontent.com/49098127/81392115-c236bd80-9150-11ea-8be7-01b85575e517.png)

**Comment by jmshal** [comment]
The Azure team pushed an update out to the Azure Function Apps infrastructure (apparently relating to the filesystem), which adds a `x-ms-privatelink-id` header, which includes a new line in it - which is invalid.

I resolved this by putting a line of code within the function handler, before the apollo library takes over:

```js
delete req.headers['x-ms-privatelink-id'];
```

(a more complete example:)

```js
const { ApolloServer, gql } = require('apollo-server-azure-functions');

const server = new ApolloServer({
  // ... all of your apollo server goodness
});

const handler = server.createHandler();

module.exports = (context, req) => {
  delete req.headers['x-ms-privatelink-id'];
  handler(context, req);
};
```

I've had a support ticket open with them for over 10 days now, and they still haven't explained why this happened, and when a patch will be pushed out. All they've said is that it was an update to the filesystem. I will update this thread with any information I get back from them.

**Comment by jmshal** [comment]
I have received an update from the support agent I was in contact with.

> Our Product Group confirmed a patch is being rolled out on the instances affected by this issue, it will be available on all instances in 2.5 weeks, we apologize for any problem this might have cause to you.

**Comment by glasser** [comment]
It sounds like this was already fixed by Azure Functions, but additionally we no longer maintain a direct integration with Azure Functions in Apollo Server 4 (instead, we have a simple stable API for building your own integrations). See https://github.com/apollographql/apollo-server/issues/6032
